File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
    
        mp = {}
        for i in range(len(nums)):
            if nums[i] in mp :
                if i > mp[nums[i]] :
                    return [mp[nums[i]],i]
                else : 
                    return [mp[i,nums[i]]]
            else :
                mp[target - nums[i]] = i
        
        return []

File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        // a ^ a = 0
        int n = nums.size();
        int ans = n;
        for (int i = 0; i < n; i++) {
            ans ^= i;
            ans ^= nums[i]; 
        }
        return ans;
        // [3,0,1]
        // init ans = 3;
        // i = 0; ans = 0
        // i = 1; ans = 1
        // i = 2; ans = 2
    }
};
